src/js/components/CosmosErrorMessage.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+class CosmosErrorMessage extends React.Component {
+  getHeader() {
+    let {header, headerClass} = this.props;
+    if (!header) {
+      return null;
+    }
+
+    return (
+      <span className={headerClass}>
+        {header}
+      </span>
+    );
+  }
+  getMessage() {
+    let {type, message} = this.props.error;
+
+    switch (type) {
+      // Specific uninstall errors
+      case 'IncompleteUninstall':
+      case 'UninstallNonExistentAppForPackage':
+        if (!message) {
+          message = 'Unable to uninstall package. Please check your connection and try again.';
```

Comment:
Is this error caused by a network issue?
If that were the case I'd expect the `type` to be defined client side. If these are client side errors then perhaps we should use constants.